compiler/rustc_transmute/src/lib.rs

```diff
@@ -171,15 +159,16 @@ mod rustc {
                 fields[field_idx].unwrap_leaf() == ScalarInt::TRUE
             };
 
-            Some(Self {
+            Some(Ok(Self {
                 alignment: get_field(sym::alignment),
                 lifetimes: get_field(sym::lifetimes),
                 safety: get_field(sym::safety),
                 validity: get_field(sym::validity),
-            })
+            }))
         }
     }
 }
 
 #[cfg(feature = "rustc")]
 pub use rustc::*;
+use rustc_span::ErrorGuaranteed;
```

Comment:
this import should live in the `rustc` module, shouldn't it? I guess why is the `Reason` enum outside of `rustc` :thinking: i would assume that we mostly need this when doing "rustc stuff"?

Reply:
Yeah, we are trying to make it so that rustc_transmute can build independently of rustc, so pulling in `ErrorGuaranteed` is not ideal. I'm going to add a FIXME for this. It's getting more and more coupled as I make fixes/improvements 😮